lib/jormungandr/src/Cardano/Wallet/Jormungandr/Api/Types.hs

```diff
@@ -73,10 +67,14 @@ import GHC.Generics
 import Servant.API
     ( Accept (..), MimeRender (..), MimeUnrender (..), ToHttpApiData (..) )
 
+import qualified Codec.Binary.Bech32 as Bech32
 import qualified Data.Aeson.Types as Aeson
 import qualified Data.ByteString.Lazy as BL
 import qualified Servant.API.ContentTypes as Servant
 
+newtype ApiAccountId = ApiAccountId { getApiAccountId :: Bech32.DataPart }
+    deriving (Eq, Show)
```

Comment:
Having a `DataPart` at this level (in the type representation) is forcing a user-facing encoding onto the users of that API type. One of the main goal of an API client is to abstract concerns such as encoding or protocol details from the client consumer. Since `ApiAccountId` is exposed as a result type, it'd be best to use a raw type that doesn't suggest any particular encoding (`ByteString` or, `ChimericAccount`).

Reply:
@KtorZ wrote:
> Since `ApiAccountId` is exposed as a result type, it'd be best to use a raw type that doesn't suggest any particular encoding.

Fair enough! I've updated `AccountId` so that the underlying representation is a `ByteString`, converting to Bech32 only within the `ToText` instance, using a customary human-readable prefix of "account" (which we can change if need to).

See: https://github.com/input-output-hk/cardano-wallet/pull/995/commits/1639467c242a5f7e769cfb47e28dd6fef32d8a8f

Reply:
Further to the above comment, `AccountId` is now encoded as a hexadecimal string, as this is what the Jormungandr REST API expects. See: https://github.com/input-output-hk/cardano-wallet/pull/995#issuecomment-553807074